route.go

```diff
@@ -74,11 +74,13 @@ func (r *Route) Match(req *http.Request, match *RouteMatch) bool {
 		return false
 	}
 
-	if match.MatchErr == ErrMethodMismatch && r.handler != nil {
+	if match.MatchErr == ErrMethodMismatch {
```

Comment:
Need more comments here, as this is a block that has changed multiple times. We should explain _why_ the code is this way - e.g.

```
// If a route matches, but the HTTP method does not, we do one of two (2) things:
// 1. Reset the match error if we find a matching method later.
// 2. Else, we override the matched handler in the event we have a possible fallback handler for that route.
//
// This is to prevent ...
```

(Feel free to suggest clearer/different documentation here).

Reply:
Have added comments. Feel free to suggest modifications